test/FilterController/index.test.js

```diff
@@ -0,0 +1,89 @@
+import React, { createClass } from 'react';
+import { configure, shallow, mount } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import chai from 'chai';
+import configureStore from 'redux-mock-store';
+import sinon from 'sinon';
+import { MemoryRouter } from 'react-router-dom';
+import Location from '@jedmao/location';
+
+import FilterController from 'Components/filter-controller';
+import ProjectsList from 'Components/containers/projects-list';
+import { a11yHelper } from '../reactA11yHelper';
+import { makeTestProject, makeTestUser } from '../helpers/models';
+import { Provider as ReduxProvider } from 'react-redux';
+import { Context as GlobalsContext } from 'State/globals';
+import * as ErrorHandlers from 'State/error-handlers';
+import * as Notifications from 'State/notifications';
+import { context as NotificationContext } from 'State/notifications';
+import { ProjectMemberContext } from 'State/project';
+import * as VisibilityContainer from 'Components/visibility-container';
+import { url } from 'inspector';
+
+chai.should();
+const middlewares = [];
+const mockStore = configureStore(middlewares);
+configure({ adapter: new Adapter() });
+
+class FakeVisibilityContainer extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+  render() {
+    return (
+      <div>{this.props.children({ isVisible: true, wasEverVisible: true })}</div>
+    );
+  }
+}
+
+describe('ProjectsList', function() {
+  describe('a11y testing', function() {
+    it('should have no errors when there are no projects', function() {
+      a11yHelper.testEnzymeComponent(<ProjectsList layout="row" projects={[]} enableFiltering />, {}, function(results) {
+        results.violations.length.should.equal(0);
+      });
+    });
+  });
+
+  it('should contain an input when filtering', function() {
+    const wrapper = mount(<ProjectsList layout="row" projects={[]} enableFiltering />);
+    return wrapper.exists('input').should.equal(true);
+  });
+
+  it('should show projects', function() {
+    const projects = [makeTestProject({ id: '1' }), makeTestProject({ id: '2' }), makeTestProject({ id: '3' })];
+    const reduxStore = mockStore({ currentUser: makeTestUser({ projects, teams: [] }) });
+    const errorStub = sinon.stub(ErrorHandlers, 'default');
+    errorStub.returns({ handleError: () => {}, handleCustomError: () => {} });
+
+    const visibilityContainerStub = sinon.stub(VisibilityContainer, 'default');
+    visibilityContainerStub.returns(FakeVisibilityContainer);
```

Comment:
The warning is caused by this. The stub is returning a reference to the class when called, so if we wrote `<VisbilityContainer />` it would evaluate not to a react component, but to the class itself. And, since classes are functions, it triggers the function value warning.

Instead, we should be using `callsFake`:

```suggestion
    visibilityContainerStub.callsFake((props) => new FakeVisibilityContainer(props));
```